services/src/unit-test/java/uk/gov/moj/sdt/services/utils/IndividualRequestsXmlParserTest.java

```diff
@@ -167,6 +191,23 @@ void testParserPerformance() throws IOException {
             assertNotNull(request.getRequestPayload(),
                     "Payload should have been populated in request#" + requestIndex);
         }
+
+        List<ILoggingEvent> logList = listAppender.list;
+        logger.detachAndStopAllAppenders();
+
+        assertTrue(verifyLog(logList,"Started parsing raw xml to extract payload for"));
+        assertTrue(verifyLog(logList,"Found match: requestId"));
+        assertTrue(verifyLog(logList,"Raw XML enhanced with namespaces"));
+    }
+
+    private static boolean verifyLog(List<ILoggingEvent> logList, String message) {
+        boolean verifyLog = false;
+        for (ILoggingEvent log : logList) {
```

Comment:
Add exit condition so that the loop terminates when message has been found.

Reply:
done